### ISSUE TITLE
Allow option to bring back fluentd json log key parsing

### ISSUE THREAD

**Issue by cloudnautique** [rancher/rancher#23646]
The behavior json parsing of log data in the K8s metadata plugin changed between Rancher's packaging of Fluentd. The capabilities:

```
merge_json_log
preserve_json_log
``` 
no longer function. 

In order to bring this capability back for users that were relying on it, we need to run it through a fluentd filter block with parser.

```
<filter **>
  @type parser
  key_name log
  reserve_data true
  <parse>
    @type json
  </parse>
</filter>
```

This should be exposed as an option in both cluster and project.

**Comment by cjellick** [comment]
@dramich to LOE

**Comment by dramich** [comment]
Based on a conversation with @cloudnautique and looking at source we can potentially insert the above blob [here](https://github.com/rancher/rancher/blob/master/pkg/controllers/user/logging/generator/templatefilter.go#L35) which should make it work the same as pre upgrade.

**Comment by dramich** [comment]
LOE 1 day

**Comment by kinolaev** [comment]
Hello @cjellick,
just want to remind that parsing json log is bad default behavior https://github.com/rancher/rancher/issues/15880#issuecomment-517671647, sо if other clients relying on it, it should be exposed only as an option (as @cloudnautique suggest).

**Comment by cjellick** [comment]
yeah, this would be the non-default behavior @kinolaev

**Comment by dramich** [comment]
master pr: https://github.com/rancher/rancher/pull/24361
2.3 pr: https://github.com/rancher/rancher/issues/23646

**Comment by izaac** [comment]
This option is not yet exposed in the UI: https://github.com/rancher/rancher/issues/24367

**Comment by izaac** [comment]
UI was completed and ready To Test

**Comment by izaac** [comment]
Rancher version: `2.2.9` `rancher/fluentd:v0.1.13` and gem version `fluent-plugin-kubernetes_metadata_filter (2.0.0)`

I was able to see how the following settings worked.

```
merge_json_log true
preserve_json_log true
```

The json sent to the fluend target (server) included the json log from an nginx workload log, preserved and merged.
```json
{
    "timestamp": "2019-12-11T18:51:35+00:00",
    "host": "x.x.x.x",
    "remote_addr": "10.42.2.0",
    "remote_user": "-",
    "body_bytes_sent": "612",
    "request_time": "0.000",
    "status": "200",
    "request": "GET / HTTP/1.1",
    "request_method": "GET",
    "http_referrer": "-",
    "http_user_agent": "curl/7.64.1",
    "log": "{ \"timestamp\": \"2019-12-11T18:51:35+00:00\", \"host\": \"x.x.x.x\", \"remote_addr\": \"10.42.2.0\", \"remote_user\": \"-\", \"body_bytes_sent\": \"612\", \"request_time\": \"0.000\", \"status\": \"200\", \"request\": \"GET / HTTP/1.1\", \"request_method\": \"GET\", \"http_referrer\": \"-\", \"http_user_agent\": \"curl/7.64.1\" }\r\n",
    "stream": "stdout",
    "docker": {
        "container_id": "8eddebb0d3c0f2ae427bd6b3a086c55e52d3e176d91f9d2c9847713d93e3f89e"
    },
    "kubernetes": {
        "container_name": "myjson",
        "namespace_name": "default",
        "pod_name": "myjson-797598fc7f-h82kb",
        "pod_id": "97ab4201-977c-4b40-a1ea-aef5559ca05a",
        "labels": {
            "pod-template-hash": "797598fc7f",
            "workload_user_cattle_io/workloadselector": "deployment-default-myjson"
        },
        "host": "izaac-v229-test1-wrkr-1",
        "master_url": "https://10.43.0.1:443/api",
        "namespace_id": "a3a372e5-61b4-489e-9f8b-4f7e2c8579b2",
        "namespace_labels": {
            "field_cattle_io/projectId": "p-ddr6x"
        }
    },
    "tag": "c-5nh89:p-ddr6x.var.log.containers.myjson-797598fc7f-h82kb_default_myjson-8eddebb0d3c0f2ae427bd6b3a086c55e52d3e176d91f9d2c9847713d93e3f89e.log",
    "log_type": "k8s_normal_container",
    "projectID": "c-5nh89:p-ddr6x"
}
```

In Rancher version `2.3-head (12/11/2019)`  I activate the those json keys are not merged even selecting the `EnableJSONParsing` checkbox in the Rancher UI and saving the settings (I see the `enableJSONParsing: true` sent to the fluentForwarder logging configuration.



I used `izaac/nginx-json-log` for testing json logging by curling port 80.

**Comment by izaac** [comment]
sending the `enableJSONParsing` manually to the API makes the log preserve and merge the json keys. So I've moved this back To Test and reopened: https://github.com/rancher/rancher/issues/24367

**Comment by izaac** [comment]
I ended up with https://github.com/rancher/rancher/issues/24545 activating cluster level logging.

**Comment by izaac** [comment]
validated working after fix of #24545 

```
2.3-head (12/27/2019)
2019/12/27 16:05:06 [INFO] Rancher version e9a549e59 is starting

master-head (12/27/2019)
2019/12/27 16:33:53 [INFO] Rancher version 189e4337c is starting
```

**Comment by soobidou** [comment]
Our corporate team updated to Rancher 2.3.2 right now, and that broke this functionality on our clusters. It's a very handy feature and I couldn't find a clear description on how to put it back with the current version we have. Any cues?
(I don't control Rancher upgrades unfortunately. And there's no such "opt-in" option in the 2.3.22 UI)
Thanks
